src/map.cpp

```diff
@@ -7505,16 +7514,55 @@ void map::draw_rough_circle(std::string type, int x, int y, int rad) {
 void map::draw_rough_circle_furn(furn_id type, int x, int y, int rad) {
     for (int i = x - rad; i <= x + rad; i++) {
         for (int j = y - rad; j <= y + rad; j++) {
-            if (rl_dist(x, y, i, j) + rng(0, 3) <= rad) {
+            if (trig_dist(x, y, i, j) + rng(0, 3) <= rad) {
                 furn_set(i, j, type);
             }
         }
     }
 }
+
 void map::draw_rough_circle_furn(std::string type, int x, int y, int rad) {
     draw_rough_circle_furn(find_furn_id(type), x, y, rad);
 }
 
+void map::draw_circle(ter_id type, double x, double y, double rad) {
+    for (int i = x - rad - 1; i <= x + rad + 1; i++) {
+        for (int j = y - rad - 1; j <= y + rad + 1; j++) {
+            if ( (x-i)*(x-i)+(y-j)*(y-j) <= rad*rad ) {
+                ter_set(i, j, type);
+            }
+        }
+    }
+}
```

Comment:
having a non-integer circle function allows for circles centered in the middle of a map. sadly none of the functions that operate on tripoints are usable helpers here. fortunately, the math for a circle is super easy.